### PR TITLE
[Util] macro safe-free

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Remove glib, check mem ptr to release allocated memory.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

**Review by gichan-jang — APPROVED** [review]
LGTM

File: src/libnnstreamer-edge/nnstreamer-edge-internal.c [hunk 1]
```diff
@@ -1080,15 +1077,15 @@ nns_edge_release_handle (nns_edge_h edge_h)
...
   eh->user_data = NULL;
-  g_free (eh->id);
-  g_free (eh->topic);
-  g_free (eh->ip);
-  g_free (eh->recv_ip);
+  SAFE_FREE (eh->id);
+  SAFE_FREE (eh->topic);
+  SAFE_FREE (eh->ip);
+  SAFE_FREE (eh->recv_ip);
   g_hash_table_destroy (eh->conn_table);
```
Comment (gichan-jang) [inline src/libnnstreamer-edge/nnstreamer-edge-internal.c:1084]: Let's free `eh->caps_str` here.
```
SAFE_FREE (eh->caps_str);
```

File: src/libnnstreamer-edge/nnstreamer-edge-common.c [hunk 2]
```diff
@@ -368,7 +367,7 @@ nns_edge_data_copy (nns_edge_data_h data_h, nns_edge_data_h * new_data_h)
     copied->data[i].data = nns_edge_memdup (ed->data[i].data,
         ed->data[i].data_len);
     copied->data[i].data_len = ed->data[i].data_len;
-    copied->data[i].destroy_cb = g_free;
+    copied->data[i].destroy_cb = free;
```
Comment (myungjoo) [inline src/libnnstreamer-edge/nnstreamer-edge-common.c:370]: Then, let's add a null check when destroy_cb is called.
Or, write a static function that does a null check here.

File: src/libnnstreamer-edge/nnstreamer-edge-internal.c (hunk 1 above)
Comment (jaeyun-jung) [reply src/libnnstreamer-edge/nnstreamer-edge-internal.c:1085]: updated

File: src/libnnstreamer-edge/nnstreamer-edge-common.c (hunk 2 above)
Comment (jaeyun-jung) [reply src/libnnstreamer-edge/nnstreamer-edge-common.c]: add new function for this.

**Pull request merged**